src/components/numeral-date/numeral-date.test.js

```diff
@@ -0,0 +1,487 @@
+    });
+
+    it.each([
+      ["left", "start"],
+      ["right", "end"],
+    ])(
+      "should render NumeralDate with labelAlign prop set to %s",
+      (labelAlign, cssValue) => {
+        CypressMountWithProviders(
+          <NumeralDateComponent labelInline labelAlign={labelAlign} />
+        );
+
+        getDataElementByValue("label")
+          .parent()
+          .should(($element) =>
+            expect($element).to.have.css("justify-content", `flex-${cssValue}`)
+          );
+      }
+    );
+
+    it.each([CHARACTERS.SPECIALCHARACTERS, CHARACTERS.DIACRITICS])(
+      "should render NumeralDate with labelHelp prop set to %s",
+      (labelHelp) => {
+        CypressMountWithProviders(
+          <NumeralDateComponent labelInline labelHelp={labelHelp} />
+        );
+
+        getDataElementByValue("question").trigger("mouseover");
+        tooltipPreview().should("have.text", labelHelp);
+      }
+    );
+
+    it.each([
+      [1, "8px"],
+      [2, "16px"],
+    ])(
+      "should render NumeralDate with labelSpacing prop set to %s",
+      (spacing, padding) => {
+        CypressMountWithProviders(
+          <NumeralDateComponent labelInline labelSpacing={spacing} />
+        );
+
+        getDataElementByValue("label")
+          .parent()
+          .should("have.css", "padding-right", padding);
+      }
+    );
+
+    it.each([
+      ["10", "90", "135px"],
+      ["30", "70", "405px"],
+      ["80", "20", "1080px"],
+    ])(
+      "should use %s as labelWidth, %s as inputWidth and render it with correct label and input width ratios",
+      (label, input, labelRatio) => {
+        CypressMountWithProviders(
+          <NumeralDateComponent
+            labelInline
+            labelWidth={label}
+            inputWidth={input}
+          />
+        );
+
+        getDataElementByValue("label")
+          .parent()
+          .should("have.css", "width", labelRatio);
+
+        getDataElementByValue("input")
+          .parent()
+          .should("have.css", "width", "58px");
+      }
+    );
+
+    it("should render NumeralDate with required prop", () => {
+      CypressMountWithProviders(<NumeralDateComponent required />);
+
+      verifyRequiredAsteriskForLabel();
+    });
+
+    it("should render NumeralDate with name prop", () => {
+      CypressMountWithProviders(
+        <NumeralDateComponent name={CHARACTERS.STANDARD} />
+      );
+
+      numeralDateComponent().should("have.attr", "name", CHARACTERS.STANDARD);
+    });
+
+    it("should render NumeralDate with disabled prop", () => {
+      CypressMountWithProviders(<NumeralDateComponent disabled />);
+
+      numeralDateInputByPosition(0)
+        .should("be.disabled")
+        .and("have.attr", "disabled");
+    });
+
+    it("should render NumeralDate with readOnly prop", () => {
+      CypressMountWithProviders(<NumeralDateComponent readOnly />);
+
+      numeralDateInputByPosition(0).and("have.attr", "readOnly");
```

Comment:
numeralDateInputByPosition(0).should("have.attr", "readOnly");

Reply:
fixed